src/index.js

```diff
@@ -129,14 +162,18 @@ let Rosnodejs = {
     const nodeOpts = options.node || {};
     rosNode = new RosNode(nodeName, rosMasterUri, nodeOpts);
 
-    return this._loadOnTheFlyMessages(options)
-      .then(_checkMasterHelper)
+    return new Promise((resolve,reject)=>{
+      this._loadOnTheFlyMessages(options)
+      .then(()=>{return _checkMasterHelper(100, options.timeout);})
       .then(Logging.initializeRosOptions.bind(Logging, this, options.logging))
       .then(Time._initializeRosTime.bind(Time, this))
-      .then(() => { return this.getNodeHandle(); })
+      .then(() => { resolve(this.getNodeHandle()); })
       .catch((err) => {
         log.error('Error during initialization: ' + err);
+        this.shutdown();
```

Comment:
this might throw additional errors depending on when the error occurred - the shutdown call in the RosNode assumes the initialization happened appropriately

Reply:
I think that's fine; the exit implementation in rosnode catches all the potential rejections that could happen while shutting down, the only thing that would happen is some extra errors in the log.

Reply:
The errors I was worried about don't seem to happen - but this line (and rejecting with the error below) do break a lot of the tests.